src/shared/Microsoft.Git.CredentialManager.Tests/HttpClientFactoryTests.cs

```diff
@@ -22,12 +26,105 @@ public void HttpClientFactory_GetClient_SetsDefaultHeaders()
         [Fact]
         public void HttpClientFactory_GetClient_MultipleCalls_ReturnsNewInstance()
         {
-            var factory = new HttpClientFactory();
+            var factory = new HttpClientFactory(Mock.Of<ITrace>(), Mock.Of<ISettings>(), new TestStandardStreams());
 
             HttpClient client1 = factory.CreateClient();
             HttpClient client2 = factory.CreateClient();
 
             Assert.NotSame(client1, client2);
         }
+
+        [Fact]
+        public void HttpClientFactory_TryCreateProxy_NoProxy_ReturnsFalseOutNull()
+        {
+            const string repoPath = "/tmp/repos/foo";
+            const string repoRemote  = "https://remote.example.com/foo.git";
+            var repoRemoteUri = new Uri(repoRemote);
+
+            var settings = new TestSettings
+            {
+                RemoteUri = repoRemoteUri,
+                RepositoryPath = repoPath
+            };
+            var httpFactory = new HttpClientFactory(Mock.Of<ITrace>(), settings, Mock.Of<IStandardStreams>());
+
+            bool result = httpFactory.TryCreateProxy(out IWebProxy proxy);
+
+            Assert.False(result);
+            Assert.Null(proxy);
```

Comment:
Is it worth verifying any fields on the proxy?

Reply:
There are no other fields on `IWebProxy` sadly (without doing some suspicious casting and/or reflection. 😢